Src/FluentAssertions/Collections/GenericCollectionAssertions.cs

```diff
@@ -285,15 +285,16 @@ public AndConstraint<TAssertions> AllBeOfType(Type expectedType, string because
     /// </param>
     public AndConstraint<TAssertions> BeEmpty(string because = "", params object[] becauseArgs)
     {
+        var singleItemArray = Subject?.Take(1).ToArray();
         Execute.Assertion
             .BecauseOf(because, becauseArgs)
             .WithExpectation("Expected {context:collection} to be empty{reason}, ")
-            .Given(() => Subject)
+            .Given(() => singleItemArray)
             .ForCondition(subject => subject is not null)
             .FailWith("but found <null>.")
             .Then
-            .ForCondition(subject => !subject.Any())
-            .FailWith("but found {0}.", Subject)
+            .ForCondition(subject => subject.Length == 0)
+            .FailWith("but found at least 1 item {0}.", singleItemArray)
```

Comment:
🤔 I've been trained that numbers up to 10 should be spelled out as "one", "two", etc. Not sure if that applies here.